internal/file/watch/watch_test.go

```diff
@@ -360,90 +362,40 @@ func Test_watch_Start(t *testing.T) {
 	tests := []test{
 		func() test {
 			ctx, cancel := context.WithCancel(context.Background())
-			return test{
-				name: "returns channel with error when watcher closed and initialize fails",
-				args: args{
-					ctx: ctx,
-				},
-				fieldsFunc: func(t *testing.T) fields {
-					t.Helper()
-					w, err := fsnotify.NewWatcher()
-					if err != nil {
-						t.Fatal(err)
-					}
-					w.Errors = make(chan error, 1)
-
-					w.Errors <- errors.New("err")
-					w.Close()
-
-					return fields{
-						w:  w,
-						eg: errgroup.Get(),
-						dirs: map[string]struct{}{
-							"vald": {},
-						},
-					}
-				},
-				afterFunc: func(t *testing.T, args args, w Watcher) {
-					t.Helper()
-					_ = w.Remove("vald")
-					defaultAfterFunc(t, args, w)
-					cancel()
-				},
-				want: want{
-					want: func() chan error {
-						ch := make(chan error, 1)
-						ch <- errors.New("err")
-						close(ch)
-						return ch
-					}(),
-					err: nil,
-				},
+			tmpDir, err := os.MkdirTemp("", "")
+			if err != nil {
+				t.Error(err)
 			}
-		}(),
+			w, err := fsnotify.NewWatcher()
+			if err != nil {
+				t.Fatal(err)
+			}
+			w.Add(tmpDir)
```

Comment:
This code seems in many times.
How about this?
```go
...
defaultWatcher := func() (string, Watcher) {
    tmpDir, err := os.MakeTmpDir("", "")
    if err != nil {
        t.Fatal(err)
    }
    w, err := fsnotify.NewWatcher()
    if err != nil {
        t.Fatal(err)
    }
    w.Add(tmpDir)
    return tmpDir, w
} 
test := []test {
    func() test {
        ctx, cancel := context.WichCancel(context.Background())
        tmpDir, w := defaultWatcher()
        return test {
            ...,
            w: w,
            ...,
        }
    }(),
    ...
}
```

Reply:
fixed